ansible/library/multi-asic_aware_module_guidelines.md

```diff
@@ -0,0 +1,71 @@
+# Guidelines for designing a customized ansible module to support multi-ASIC
+
+PR https://github.com/Azure/SONiC/pull/644 introduced the HLD to support multi ASIC. In the future, multi DUT or Chassis will be supported by SONiC as well. Some of the customized ansible modules need to be updated to support testing of the upcoming new architectures. This document tries to propose some guidelines for designing customized ansible modules that need to deal with multi-ASIC. The idea is to have a clean and easy to use interface for calling these ansible modules in scripts testing multi DUT and multi ASIC system. Meanwhile, the ansible modules need to maintain backward compatibility for single DUT and single ASIC testing.
+
+## Guidelines
```

Comment:
Are these guidelines or hard requirements? I think we should be explicit, otherwise some modules may not do what we expect them to do.

Reply:
Good catch! You are right, these should be requirements. I'll update the wording.